### PR TITLE
Remove script from theme.inc

### PR DESCRIPTION
**Pull request by 70ray**
Removes embedded javascript from theme.inc.

Sandbox at: https://www.pgdp.org/~rp31/c.branch/theme_script

### CODE REVIEW

File: scripts/navbar_menu.js [hunk 1]
```diff
@@ -46,6 +43,13 @@ function hideSiteSearch() {
 }
 
 window.addEventListener('DOMContentLoaded', function() {
+    const menuSelectors = document.getElementsByClassName('menu-selector');
+    for(const menuSelector of menuSelectors) {
```
Comment (cpeel) [inline scripts/navbar_menu.js:47]:
```suggestion
    for (const menuSelector of menuSelectors) {
```

File: scripts/navbar_menu.js [hunk 2]
```diff
@@ -54,11 +58,24 @@ window.addEventListener('DOMContentLoaded', function() {
...
                 event.preventDefault();
                 showSiteSearch();
             }
         });
     }
+
+    // if the page has a language selector, listen for a change
+    let langSelector = document.getElementById('langform-lang');
+
+    if(langSelector) {
```
Comment (cpeel) [inline scripts/navbar_menu.js:75]:
```suggestion
    if (langSelector) {
```

File: pinc/theme.inc [hunk 3]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
     output_html_footer();
 }
 
+abstract class actionIcon
```
Comment (cpeel) [inline pinc/theme.inc:122]: PHP class names should be PascalCase to match all of our other class names.

File: pinc/theme.inc [hunk 4]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+    }
+
+    public function draw()
+    {
+        echo "<div class='icon-menu-item'>";
+        $this->draw_action();
+        echo "</div>";
+    }
+
+    public function draw_icon()
```
Comment (cpeel) [inline pinc/theme.inc:140]:
```suggestion
    protected function draw_icon()
```

File: pinc/theme.inc [hunk 5]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+    {
+        echo "<i aria-hidden='true' class='$this->icon' title='$this->name'></i>";
```
Comment (cpeel) [inline pinc/theme.inc:142]: `this->name` needs an `attr_safe()`

File: pinc/theme.inc [hunk 6]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+
+    public function __construct(string $name, string $icon, string $url)
+    {
+        $this->url = $url;
+        parent::__construct($name, $icon);
+    }
+
+    public function draw_action()
+    {
+        echo "<a href='$this->url' aria-label='$this->name'>";
```
Comment (cpeel) [inline pinc/theme.inc:166]: Ditto `attr_safe()` for both of these.

File: pinc/theme.inc [hunk 7]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+        $this->draw_icon();
+        echo "</a>";
+    }
+}
+
+class searchIcon extends actionIcon
+{
+    public function draw_action()
+    {
+        echo "<div tabindex='0' id='site_search' aria-label='$this->name'>";
```
Comment (cpeel) [inline pinc/theme.inc:176]: `attr_safe()` here too.

File: pinc/theme.inc [hunk 8]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+    public function draw_icon()
+    {
+        echo "<i aria-hidden='true' class='$this->icon' title='$this->name'></i>";
+        $this->draw_extra();
+        echo "<br><span class='icon-menu-item-name'>$this->name</span>";
+    }
+
+    abstract public function draw_action();
+
+    public function draw_extra()
```
Comment (cpeel) [inline pinc/theme.inc:149]:
```suggestion
    abstract protected function draw_action();

    protected function draw_extra()
```

File: pinc/theme.inc (hunk 3 above)
Comment (cpeel) [reply pinc/theme.inc:122]: Also `MenuIcon` might be a better name since it's in a `icon-menu`-wrapped div.

File: pinc/theme.inc [hunk 9]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+    {
+    }
+}
+
+class linkIcon extends actionIcon
```
Comment (cpeel) [inline pinc/theme.inc:154]: `MenuLinkIcon`?

File: pinc/theme.inc [hunk 10]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+
+    public function draw_action()
+    {
+        echo "<a href='$this->url' aria-label='$this->name'>";
+        $this->draw_icon();
+        echo "</a>";
+    }
+}
+
+class searchIcon extends actionIcon
```
Comment (cpeel) [inline pinc/theme.inc:172]: `MenuSearchIcon`?

File: pinc/theme.inc [hunk 11]
```diff
@@ -119,6 +119,78 @@ function output_footer($nameofpage, $show_statsbar = true)
...
+{
+    public function draw_action()
+    {
+        echo "<div tabindex='0' id='site_search' aria-label='$this->name'>";
+        $this->draw_icon();
+        echo "</div>\n";
+    }
+}
+
+class inboxIcon extends linkIcon
```
Comment (cpeel) [inline pinc/theme.inc:182]: `MenuInboxIcon`?

**Review by cpeel — COMMENTED** [review]
That's a lot of abstraction but I think it does make things clearer.

File: scripts/navbar_menu.js (hunk 1 above)
Comment (chrismiceli) [reply scripts/navbar_menu.js:47]: we have a linter which should enforce some of these suggestions I believe:
https://github.com/DistributedProofreaders/dproofreaders/blob/master/.eslintrc.json
If not, we can definitely codify our desired format.

Comment (cpeel) [reply scripts/navbar_menu.js:47]: It appears to have [passed the eslint linter](https://github.com/DistributedProofreaders/dproofreaders/actions/runs/10391438980/job/28774349190?pr=1300)? Wonder if something's not configured quite right.

Comment (chrismiceli) [reply scripts/navbar_menu.js:47]: Let me investigate. We may just not have too many format rules configured.

**Comment by 70ray** [comment]
> That's a lot of abstraction but I think it does make things clearer.

I did this because it didn't seem very clear where to put the id in the original but, looking at it again, I think it should be fairly easy. So it would perhaps be better to keep closer to the original for now. I'll make a different PR.

**Comment by cpeel** [comment]
I really don't mind the abstraction you have in place if we can update some of the things I identified in the PR. Whatever you'd like to do.

**Comment by 70ray** [comment]
New revised PR opened.

**Pull request closed without merging**